edb/edgeql/compiler/func.py

```diff
@@ -328,11 +328,18 @@ def compile_operator(
             )
 
         if not matched:
+            if op_name == 'std::IF':
+                hint = ("The IF and ELSE result clauses must be of compatible "
+                        "types, while the condition clause must be "
+                        "'std::bool'.")
+            else:
+                hint = ('Consider using an explicit type cast or a conversion '
```

Comment:
I don't think these are mutually exclusive.  This is still a valid hint in the 'std::IF' case.

Reply:
Right, I'll merge them for IF.